db/db_impl.cc

```diff
@@ -4759,18 +4759,18 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
       if (!cfd->mem()->IsEmpty()) {
         // We only consider active mem table, hoping immutable memtable is
         // already in the process of flushing.
-        size_t cfd_size = cfd->mem()->ApproximateMemoryUsage();
-        if (largest_cfd == nullptr || cfd_size > largest_cfd_size) {
-          largest_cfd = cfd;
-          largest_cfd_size = cfd_size;
+        uint64_t o = cfd->OldestLogToKeep();
```

Comment:
This won't work well if all writes have WAL disabled, right? OldestLogToKeep() will have the same value in all CFs, and this loop will pick up the oldest CF; correct?

Would it make sense to pick CF with the smallest (nonzero) GetFirstSequenceNumber() instead? Seems that it would have the same behavior as OldestLogToKeep() when WAL is used, but will also work when WAL is not used.